microcloud/test/includes/microcloud.sh

```diff
@@ -933,7 +935,7 @@ setup_system() {
 
     # Install the snaps.
     lxc exec "${name}" -- apt-get update
-    if [ -n "${CLOUD_INSPECT:-}" ]; then
+    if [ -n "${CLOUD_INSPECT:-}" ] || [ "${SNAPSHOT_RESTORE}" = 0 ]; then
```

Comment:
If I followed properly, this isn't needed anymore now that #301 is closed there is no need to have the test automatically issue `zpool` commands.

Reply:
#301 just introduced a sleep to the `zpool destroy` call here: https://github.com/canonical/microcloud/blob/4f47db906f5846928fc56e8082e55c9a7d78f117/microcloud/test/includes/microcloud.sh#L524

In the non-snapshot case, we still need to make the call because sometimes LXD won't clean up the zpool properly, and you'll start another test with the disk in use.

Reply:
Right. I take it that `blkdiscard --force` or `wipefs -a -f` is not enough?

Reply:
Not if `zfs` won't let go of the disk. You'll get a `Device or resource busy`

Reply:
Frustratingly, `wipefs -a` will fail with the error but `wipefs -a -f` will return successfully, but the disk will still have partitions and be in use.

Reply:
Hmm, I'd like to try that. Might help with another issue I've seen with MicroCeph. 

I added `microceph.ceph osd set noup` because occasionally the OSDs would bring themselves back up mid-teardown, but if we just went ahead with the teardown of microceph, then we would nuke ceph, but the disk would remain in use.

I couldn't find anything obvious in `lsof`, `ps` or `/proc/self/mountinfo` that mentioned the disk was in use. 

Manually deleting disk records with `echo 1 > /sys/block/{disk_id}/device/delete` and rescanning with `echo '- - -' > /sys/class/scsi_host0/scan` would instead pick up 2 disks for the same disk. So if you had `sdb` grabbed by `ceph`, you after rescanning you would have `sdb` with partitions and `sdc` without any.

Reply:
> In the non-snapshot case, we still need to make the call because sometimes LXD won't clean up the zpool properly, and you'll start another test with the disk in use.

@masnax is this a bug in LXD? Logged an issue for it?

Reply:
I think it's a bug with zfs. LXD appears to attempt to destroy the zpool, but it very rarely doesn't work. 

I can log an issue but I don't really have much of a reproducer.

Reply:
what error do you get when doing `lxc storage delete`?

Reply:
None. The pool is gone from LXD, and there's no error reported but `zpool list` still reports a local zpool.

Reply:
I should mention, `zpool destroy mypool` will return successfully but not destroy the pool in these cases, it needs `-f`. We can add a contingency to LXD thar tries to delete the pool with `-f` if the regular delete doesn't work, but I'm not 100% clear on the implications.

Reply:
`-f` unmounts data sets, so suggests its using a lazy unmount approach if not passed. LXD should probably pass `-f` then to ensure and fail if there are still mounted data sets (this should help identity any areas were mounts are being left over).